packages/venia-concept/src/components/ProductFullDetail/ProductFullDetail.js

```diff
@@ -49,18 +51,85 @@ class ProductFullDetail extends Component {
         addToCart: func.isRequired
     };
 
-    state = { quantity: 1 };
+    state = { optionSelections: new Map(), quantity: 1 };
 
     setQuantity = quantity => this.setState({ quantity });
 
-    addToCart = () =>
-        this.props.addToCart({
-            item: this.props.product,
-            quantity: this.state.quantity
-        });
+    addToCart = () => {
+        const { props, state } = this;
+        const { optionSelections, quantity } = state;
+        const { addToCart, product } = props;
+        const {
+            __typename: productType,
+            configurable_options,
+            variants
+        } = product;
+
+        const payload = {
+            item: product,
+            productType,
+            quantity
+        };
+
+        if (productType === 'ConfigurableProduct') {
+            const options = Array.from(optionSelections, ([id, value]) => ({
+                option_id: id,
+                option_value: value
+            }));
+
+            const item = variants.find(({ product: variant }) => {
+                for (const [id, value] of optionSelections) {
+                    const { attribute_code: code } = configurable_options.find(
+                        ({ attribute_id }) => attribute_id === id
+                    );
+
+                    if (variant[code] !== value) {
+                        return false;
```

Comment:
This of course is the code we would rather the server do. It's `O(2^n)`, growing exponentially slower with each additional option and variant. Super dangerous. We should add a comment noting that we're aware of this and it's temporary.

If it's perversely slow for the time being, we could optimize this by lazy-building objects instead of arrays. We'll cross that bridge when we come to it.